### PR TITLE
Remove the non-deterministic module export modification in native.ts

### PR DESCRIPTION
**Pull request by Eximius**
In rare cases (notably jest), if it reloads the module cache for every test-suite, all test-suites that are run after the first one will have the `methods` dict have the values {send: undefined, received: undefined, ...}, because the ./native module itself is reloaded, however the node-gyp-build module is not (hence has the aforementioned dict keys of Socket.prototype deleted => undefined).

Instead of spelunking into figuring out how to make the previous very-non-standard semantics work, the changes I introduce seem like a simpler and more robust solution.

### CODE REVIEW

**Comment by lesmoutonssauvages** [comment]
Thanks @Eximius , i wasn't crazy

**Comment by rolftimmermans** [comment]
I'd be happy to merge this; but there seem to be some unrelated changes included. Could you update the PR so only the relevant code bits are modified?

**Comment by Eximius** [comment]
I can remove the bumps, but the other changes would seemingly help in general?

**Comment by rolftimmermans** [comment]
If there are any other issues you want to address feel free to open a separate pull request for them. Without any description it's not obvious why the other changes are needed.

**Comment by lesmoutonssauvages** [comment]
Hi @rolftimmermans  and @Eximius, any new on this PR ? May i help to split commits?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "zeromq",
-  "version": "6.0.0-beta.6",
+  "name": "@eximius/zeromq",
+  "version": "6.0.0-beta.11",
```
Comment (aminya) [inline package.json:3]: Please revert

File: script/build.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (aminya) [inline script/build.sh:1]: Isn't the script sh compliant?

File: src/socket.cc [hunk 3]
```diff
@@ -216,7 +216,7 @@ void Socket::WarnUnlessImmediateOption(int32_t option) const {
 
     if (immediate.count(option) != 0) return;
     if (endpoints == 0 && state == State::Open) return;
-    Warn(Env(), "Socket option will not take effect until next connect/bind.");
+    //Warn(Env(), "Socket option will not take effect until next connect/bind.");
```
Comment (aminya) [inline src/socket.cc:219]: Why commenting this out?

File: src/util.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * This function is used to remove the given methods from the given socket_prototype
+ * to make the relevant socket types have only their relevant methods.
+ * @param socketPrototype
+ * @param methods
+ */
+export function allowMethods(socketPrototype: any, methods: SocketMethods[]) {
+  const toDelete = ["send", "receive", "join", "leave"] as SocketMethods[]
+  for (const method of toDelete) {
+    if (methods.indexOf(method) != -1) delete socketPrototype[method]
```
Comment (aminya) [inline src/util.ts:13]:
```suggestion
    if (methods.indexOf(method) !== -1) { 
      delete socketPrototype[method]
    }
```

**Review by aminya — CHANGES_REQUESTED** [review]

File: src/socket.cc [hunk 5]
```diff
@@ -218,7 +218,7 @@ void Socket::WarnUnlessImmediateOption(int32_t option) const {
 
     if (immediate.count(option) != 0) return;
     if (endpoints == 0 && state == State::Open) return;
-    Warn(Env(), "Socket option will not take effect until next connect/bind.");
+    // Warn(Env(), "Socket option will not take effect until next connect/bind.");
```
Comment (aminya) [inline src/socket.cc:221]:
```suggestion
    Warn(Env(), "Socket option will not take effect until next connect/bind.");
```

**Review by aminya — APPROVED** [review]

**Pull request merged**